src/game/common/userpreferences.cpp

```diff
@@ -61,8 +61,8 @@ bool UserPreferences::Write()
     FILE *fp = fopen(m_filename.Str(), "w");
 
     if (fp != nullptr) {
-        for (auto it = begin(); it != end(); ++it) {
-            fprintf(fp, "%s = %s\n", it->first.Str(), it->second.Str());
+        for (auto &it : *this) {
```

Comment:
What is `it` ? Can this have a better name? If not please don't use `auto` type. It is very difficult to read from review what is going on here.